src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseTestCaseResource.java

```diff
@@ -41,31 +41,47 @@ public class ProgrammingExerciseTestCaseResource {
 
     private final UserRepository userRepository;
 
+    private final ParticipationRepository participationRepository;
+
     public ProgrammingExerciseTestCaseResource(ProgrammingExerciseTestCaseRepository programmingExerciseTestCaseRepository,
             ProgrammingExerciseTestCaseService programmingExerciseTestCaseService, ProgrammingExerciseService programmingExerciseService,
-            ProgrammingExerciseRepository programmingExerciseRepository, AuthorizationCheckService authCheckService, UserRepository userRepository) {
+            ProgrammingExerciseRepository programmingExerciseRepository, AuthorizationCheckService authCheckService, UserRepository userRepository,
+            ParticipationRepository participationRepository) {
         this.programmingExerciseTestCaseRepository = programmingExerciseTestCaseRepository;
         this.programmingExerciseTestCaseService = programmingExerciseTestCaseService;
         this.programmingExerciseService = programmingExerciseService;
         this.programmingExerciseRepository = programmingExerciseRepository;
         this.authCheckService = authCheckService;
         this.userRepository = userRepository;
+        this.participationRepository = participationRepository;
     }
 
     /**
-     * Get the exercise's test cases for the the given exercise id.
+     * Get the exercise's test cases for the given exercise id.
+     * If the accessor is a student all invisible test cases are filtered and the hidden test cases
      *
-     * @param exerciseId of the the exercise.
+     * @param exerciseId of the exercise.
      * @return the found test cases or an empty list if no test cases were found.
      */
     @GetMapping(Endpoints.TEST_CASES)
-    @PreAuthorize("hasRole('TA')")
+    @PreAuthorize("hasRole('USER')")
     public ResponseEntity<Set<ProgrammingExerciseTestCase>> getTestCases(@PathVariable Long exerciseId) {
```

Comment:
The test class for this resource did not change as far as I can see. Are there tests to make sure that this endpoint does not reveal tests unintentionally to students with the changed accessibility?

Reply:
This was a leftover because I tried to fix issue #4452 in this PR as well.
But I could not find an easy way without refactoring too much. So this will have to be postponed to a future PR.

The endpoint is now back to the version before this PR